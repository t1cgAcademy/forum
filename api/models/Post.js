const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const PostSchema = Schema({
  courseID: {
    type: Number,
    required: true
  },
  parentID: {
    type: String,
    required: false,
  },
  date: {
    type: Date,
    required: true,
  },
  author: {
    type: String,
    required: true,
  },
  summary: {
    type: String,
    required: false,
  },
  content: {
    type: String,
    required: true,
  },
});



module.exports = Post = mongoose.model('posts', PostSchema);
