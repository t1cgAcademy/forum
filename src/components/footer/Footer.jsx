import React from 'react';
import './footer.css';

export default function Footer() {
  return (
    <footer className="bg-dark text-white mt-5 p-4 text-center">
      Copyright &copy; {new Date().getFullYear()} T1CG Forum
    </footer>
  );
}
