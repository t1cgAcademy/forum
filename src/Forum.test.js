import React from 'react';
import Footer from './components/footer/Footer';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Post from './components/post/Post';

// Configure the adapter
Enzyme.configure({ adapter: new Adapter() });

// Snapshot test
it('renders Footer correctly', () => {
  const footer = renderer.create(<Footer />).toJSON();
  expect(footer).toMatchSnapshot();
});

// Test if renders
describe('Post component', () => {
  test('renders', () => {
    const wrapper = shallow(<Post />);

    expect(wrapper.exists()).toBe(true);
  });
});

// Test handleChange event
it('should call onChange prop', () => {
  const handleChangeMock = jest.fn();
  const event = {
    preventDefault() {},
    target: { value: 'the-value' }
  };
  const component = Enzyme.shallow(
    <Post
      handleChange={handleChangeMock}
      clearPost={jest.fn()}
      submitNewPost={jest.fn()}
    />
  );
  component
    .find('input')
    .at(0)
    .simulate('change', event.target.value);
  expect(handleChangeMock).toBeCalledWith('the-value');
});
