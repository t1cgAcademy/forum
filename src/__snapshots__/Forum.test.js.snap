// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<footer
  className="bg-dark text-white mt-5 p-4 text-center"
>
  Copyright Â© 
  2019
   T1CG Forum
</footer>
`;
